src/Utils/output_writer_diagnostic_utils.jl

```diff
@@ -8,22 +8,16 @@ using Oceananigans: AbstractOutputWriter, AbstractDiagnostic
 #####
 
 defaultname(::AbstractDiagnostic, nelems) = Symbol(:diag, nelems+1)
+
 defaultname(::AbstractOutputWriter, nelems) = Symbol(:writer, nelems+1)
 
 const DiagOrWriterDict = OrderedDict{S, <:Union{AbstractDiagnostic, AbstractOutputWriter}} where S
 
-function push!(container::DiagOrWriterDict, elem)
-    name = defaultname(elem, length(container))
-    container[name] = elem
-    return nothing
-end
+push!(container::DiagOrWriterDict, elem) = (container[defaultname(elem, length(container))] = elem; nothing)
```

Comment:
I find this a little harder to understand because of the nested calls. (But I love all the other changes on this PR!)

Reply:
yes, I do prefer it as it was before myself
it's no other change other than line merging, right?